services/api/service.go

```diff
@@ -1453,24 +1489,36 @@ func (api *RelayAPI) handleSubmitNewBlock(w http.ResponseWriter, req *http.Reque
 	}
 
 	// recalculate top bid
-	err = api.redis.UpdateTopBid(payload.Slot(), payload.ParentHash(), payload.ProposerPubkey())
+	topBidValue, err := api.redis.UpdateTopBid(payload.Slot(), payload.ParentHash(), payload.ProposerPubkey())
 	if err != nil {
 		log.WithError(err).Error("could not compute top bid")
 		api.RespondError(w, http.StatusInternalServerError, err.Error())
 		return
 	}
 
 	// after top bid is updated, the bid is eligible to win the auction.
-	log = log.WithField("timestampAfterUpdateTopBid", time.Now().UnixMilli())
 	eligibleAt = time.Now().UTC()
 
+	log = log.WithFields(logrus.Fields{
+		"timestampAfterUpdateTopBid": time.Now().UTC().UnixMilli(),
+		"newTopBidValue":             topBidValue,
+	})
+
+	// Update in-memory top bid value (if needed)
+	if topBidValue != nil || localTopBidValue.Cmp(topBidValue) < 1 {
+		api.builderTopBidValueLock.Lock()
+		api.builderTopBidValue[payload.ParentHash()] = topBidValue
```

Comment:
`api.builderTopBidValue[payload.ParentHash()]` could have changed in the meanwhile, and no longer be `localTopBidValue`

Reply:
removed the comparison in the latest commit. after updating the bid in `UpdateTopBid`, the local value should _always_ be overwritten.